src/main/java/com/dumbdogdiner/stickyapi/bukkit/util/PlayerSelector.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2021 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.bukkit.util;
+
+import com.dumbdogdiner.stickyapi.common.util.MathUtil;
+import org.bukkit.Bukkit;
+import org.bukkit.GameMode;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.permissions.ServerOperator;
+import org.bukkit.util.Vector;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * <p>
+ * Util class for selecting players fulfilling a given criterion
+ * </p>
+ */
+public class PlayerSelector {
+
+    /**
+     * <p>
+     * Get the offset between two {@link org.bukkit.Location} objects
+     * in a 3d environment using {@link Vector} objects.
+     * </p>
+     *
+     * @param pointA first location vector
+     * @param pointB second location vector
+     * @return the offset (distance) between the given locations
+     */
+    public static double offset(Vector pointA, Vector pointB) {
+        return pointA.clone().subtract(pointB).length();
+    }
+
+    /**
+     * <p>
+     * This Predicate will return true if the player is not in gamemode spectator
+     * </p>
+     */
+    public static Predicate<Player> NOT_SPECTATING = player -> player.getGameMode() != GameMode.SPECTATOR;
+
+    /**
+     * <p>
+     * This Predicate will return true if the player is in gamemode survival
+     * </p>
+     */
+    public static Predicate<Player> IN_SURVIVAL = player -> player.getGameMode() == GameMode.SURVIVAL;
+
+    /**
+     * <p>
+     * This Predicate will return true if the player is in gamemode creative
+     * </p>
+     */
+    public static Predicate<Player> IN_CREATIVE = player -> player.getGameMode() == GameMode.CREATIVE;
```

Comment:
I mean if we're adding all these, might as well add IS_SPECTATING, IS_ADVENTURE_MODE, etc